File: microservice-product/src/main/java/com/products/domain/entity/PriceType.java
package com.products.domain.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Data @NoArgsConstructor @AllArgsConstructor
@Entity @Table(name = "product_price_type")
public class PriceType {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotBlank
    private String name;

    @Getter
    public static enum Value {
        COMMON(1, "common"),
        OFFER(2, "offer"),
        BLACK_FRIDAY(3, "black_friday");

        private final int id;
        private final String name;

        Value(int id, String name) {
            this.id = id;
            this.name = name;
        }

        public static Value fromId(int id) {
            return switch (id){
                case 1 -> COMMON;
                case 2 -> OFFER;
                case 3 -> BLACK_FRIDAY;
                default -> null;
            };
        }
    }
}

File: microservice-pricing/src/main/kotlin/com/pricing/domain/entity/PriceType.kt
package com.pricing.domain.entity

enum class PriceType(val id: Int, val label: String) {
    COMMON(1, "Common"),
    OFFER(2, "Offer"),
    BLACK_FRIDAY(3, "Black Friday");

    companion object {
        fun getById(id: Int): PriceType {
            for (priceType in PriceType.entries)
                if (priceType.id == id) return priceType
            throw IllegalArgumentException("Invalid Price Type with id $id")
        }
    }
}
